devito/data/data.py

```diff
@@ -174,9 +175,9 @@ def __str__(self):
         return super(Data, self._local).__str__()
 
     @_check_idx
-    def __getitem__(self, glb_idx, comm_type):
+    def __getitem__(self, glb_idx, comm_type, _gather=None):
```

Comment:
variable name == method name ?

Reply:
Yeah, the whole variable possibly needs an overhaul.

So I'm thinking should it be allowed to be a `tuple`? i.e. you can gather simultaneously on multiple ranks. I'm not sure how often users would want this though + it adds a decent bit of complexity to the routine to save users (probably?) a couple of lines of code.

So would only be worth it if users are going to frequently do something along the lines of
```
a = f.data_gather(rank='all')
```